src/ucp/stream/stream_send.c

```diff
@@ -109,44 +148,48 @@ UCS_PROFILE_FUNC(ucs_status_ptr_t, ucp_stream_send_nb,
         goto out;
     }
 
-    if (ucs_likely(UCP_DT_IS_CONTIG(datatype)) &&
-        ucp_memory_type_cache_is_empty(ep->worker->context)) {
-        length = ucp_contig_dt_length(datatype, count);
-        if (ucs_likely((ssize_t)length <= ucp_ep_config(ep)->am.max_short)) {
-            status = UCS_PROFILE_CALL(ucp_stream_send_am_short, ep, buffer,
-                                      length);
-            if (ucs_likely(status != UCS_ERR_NO_RESOURCE)) {
-                UCP_EP_STAT_TAG_OP(ep, EAGER);
-                ret = UCS_STATUS_PTR(status); /* UCS_OK also goes here */
-                goto out;
+    if (ucp_memory_type_cache_is_empty(ep->worker->context)) {
+        attr_mask = param->op_attr_mask &
+                    (UCP_OP_ATTR_FIELD_DATATYPE | UCP_OP_ATTR_FLAG_NO_IMM_CMPL);
```

Comment:
code duplication, maybe add a macro?

Reply:
simple code, macro will make it complicated